x/staking/legacy/v036/types.go

```diff
@@ -30,7 +29,7 @@ type (
 
 	Validator struct {
 		OperatorAddress         sdk.ValAddress          `json:"operator_address" yaml:"operator_address"`
-		ConsPubKey              crypto.PubKey           `json:"consensus_pubkey" yaml:"consensus_pubkey"`
+		ConsPubKey              cryptotypes.PubKey      `json:"consensus_pubkey" yaml:"consensus_pubkey"`
```

Comment:
should this stay tmcrypto.PubKey and the migration script turn it to sdkcrypto.Pubkey. 

This question would be the same for the different versions as well.

Reply:
The "trick" used here is that `cryptotypes.PubKey` has the same Amino serialization as `crypto.PubKey`. Since migrate scripts only really cares serialization and deserialization, it doesn't matter which intermediate Golang types we use.

I added 2 tests in 19be85470fec54a5f9dc34be848ed320678108e6 to reflect this.

Reply:
ah okay. Even if you modify the interface? I am not sure what plans you have for the pubkey interface but with amino I would guess it may change?

Reply:
> Even if you modify the interface?

I was under the impression that the interface itself is never encoded in the final JSON/binary. Only the concrete type is, under the given string in `RegisterConcrete`.

Say in BaseAccount, whether the intermediate interface is `cryptotypes.PubKey` or `tmcrypto.PubKey`, the amino JSON will always be:
```
{
  "type": "tendermint/PubKeyEd25519", // registered via RegisterConcrete
  "value": "..."
}
```

Can you confirm?

Reply:
Yes i believe that part will be the same, ah sorry i was thinking of binary encoding. This is where interface get encoded in amino but this isnt an issue in sdk anymore.